rikctl/src/cli/resource/mod.rs

```diff
@@ -20,3 +21,17 @@ pub enum GetMultipleResource {
     /// List workloads,
     Workload(GetMultipleWorkload),
 }
+
+/// Trait which defines how resources should be displayed
+trait DisplayResource<T = Self>
+where
+    T: Sized,
+{
```

Comment:
Tell me if i am wrong but i think it is equivalent to just `trait DisplayResource: Sized`

Reply:
It could be true it there wasn't `<T = Self>`, if it wasn't there we would have t describe `T` whenever we implement the trait, which would lead to something similar:

```rust
impl DisplayResource<Workload> for Workload {
   ...
}
```

Here, we end up with something clearer

```rust
impl DisplayResource for Workload {
   ....
}
```

Reply:
I think that you are trying `de me couiller` here.
In both case you the code said that any implementation of this feature must be a structure that has a known size at compile time.
If you used T in the method defined by the line then maybe it would have been legit but you don't do that here :eyes: